src/components/color_picker/color_picker_entry.xml

```diff
@@ -0,0 +1,20 @@
+<templates>
+  <t t-name="o-spreadsheet-ColorPickerEntry" owl="1">
+    <div class="o-dropdown o-color-picker-entry" t-att-disabled="props.disabled">
```

Comment:
I think that the o-dropdown, and o-tools stuff should  not be part of this component. because the rules for o-dropdown and o-tools are defined anyways in topBar and sidePanel .

This component should only contain a div to set the title (by the way it should be translated befiore getting passed as props) and the div would contain the span etc..; 

```xml
<div clas="o-color-picker-stuff">
    <span ... />
</div>
<ColorPicker ... />
```

Reply:
@rrahir For the title translation, I use a getter in `ColorPickerEntry` component to translate it. Is there any difference between translating before passing as props and translating when getting it?

Reply:
Yes :) The translation flow is such that we cannot reliably translate any variable, because we don't know yet what the term will be beforehand- the props might be anything ^^^

So how to translations work in Odoo:  We have a module that will parse all files in the repository and apply different rules : 

- for xml files: it will capture all text in standard HTML tags and attributes: so  attributes like ``title="text"`,  `<span> Text2</span>` -> text and Text2  will be captured. but not parametric stuff `t-att-title="text4"` will not be captured
- for python files: it will capture the arguments of function `_(...)`
- for js files, it will capture the arguments of function `_t(...)` and `_lt(...)`

Each captured term is then exported in the POT files, which in turn will be translated in different languages. the translations are stored in PO files.
When calling the actual translation inside the code, we will try to match the string inside the translation function (_t) against the terms that are translated in the POT/PO file and see if we find a match.

This approach means that the captured terms need to be actual string! If we try to translate a variable, then we **might** sometimes find a match, but since we  never explicitely wrote in the code that we wanted our specific term to be translated, it's more likely that we won't.

note: It supports strings with placeholders, like `I am %s years old`, which can become `j'ai %s ans`, `Ik ben %s jaar oud` , etc